scqubits/tests/diag_tests.py

```diff
@@ -0,0 +1,281 @@
+import numpy as np
+import importlib
+import warnings
+import pytest
+import scqubits as scq
+import scipy as sp
+from scqubits import Fluxonium, Transmon, HilbertSpace
+
+diag_methods = scq.DIAG_METHODS.keys()
+
+def _library_installed(library):
+    try:
+        importlib.import_module(library)
+        return True
+    except ImportError:  # `ModuleNotFoundError` is a subclass of `ImportError`. Here, the general class of errors are checked.
+        return False
+
+def _get_library_methods(library, evals=True):
+    library_diag_methods = [
+        method for method in diag_methods
+        if method.split('_')[1] == library
+        and method.split('_')[0] == ('evals' if evals else 'esys')
+        and '_shift-inverse' not in method
```

Comment:
`_shift-inverse` type methods are omitted from all tests since it seems they need to be investigated in more detail to see why do they not agree with the default method. I'll try to see if I can find the reason.